core/src/main/scala/org/apache/spark/scheduler/SparkListener.scala

```diff
@@ -84,6 +85,14 @@ case class SparkListenerBlockManagerRemoved(time: Long, blockManagerId: BlockMan
 @DeveloperApi
 case class SparkListenerUnpersistRDD(rddId: Int) extends SparkListenerEvent
 
+@DeveloperApi
+case class SparkListenerExecutorAdded(executorId: String, executorDetails: ExecutorDetails)
+  extends SparkListenerEvent
+
+@DeveloperApi
+case class SparkListenerExecutorRemoved(executorId: String, executorDetails: ExecutorDetails)
```

Comment:
does the executor removed event need to contain executor details? The listener already listens for executor added events, which contain the same details